api/envoy/extensions/http/header_validators/envoy_default/v3/header_validator.proto

```diff
@@ -115,4 +115,7 @@ message HeaderValidatorConfig {
   // Envoy will respond with 400 to requests with disallowed methods.
   // By default methods with arbitrary names are accepted.
   bool restrict_http_methods = 3;
+
+  // Determines whether headers that contain underscores in their names should be accepted or not.
```

Comment:
headers -> header keys?
Also can this be a boolean?  it's an enum in the protocol options

Reply:
For what it's worth these are technically "field names", or "header names" - see RFC 9110.

Reply:
@yanavlasov the current [`headers_with_underscores_action`](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/protocol.proto#envoy-v3-api-field-config-core-v3-httpprotocoloptions-headers-with-underscores-action) config is an enum with 3 options: allow, drop, or reject. The current UHV setting is simply a boolean to allow (default) or reject the request if a header name contains an underscore. Is a boolean sufficient or do you think we need to have UHV mirror the current implementation?

Reply:
Oh, I see. Yes, we should have the same behavior as before. Can you replace the bool with the same enum you have linked, please?